### ISSUE TITLE
Custom source not found (in version 5.0.9)

### ISSUE THREAD

**Issue by tonirenner** [sebastianfeldmann/phpbu#136]
Hi,

i added a custom source in my bootstrap file like this

```php
<?php
require_once __DIR__ . '/../../vendor/autoload.php';

phpbu\App\Factory::register(
	'source',
	'<alias>',
	'<classname>'
);
```

and run phpbu with --bootstrap=bootstrap.php.

Works with phpbu 5.0.8, but in version 5.0.9 i got an error "unknown source".

Looks like issue https://github.com/sebastianfeldmann/phpbu/issues/124.

Can you verify, please?

Thanks,
Toni

**Comment by sebastianfeldmann** [comment]
So this is working for me

`backup/bootstrap.php`
```php
<?php
use phpbu\App;
class MySource implements App\Backup\Source
{
    public function setup(array $conf = [])
    {
        // foo
    }

    public function backup(\phpbu\App\Backup\Target $target, \phpbu\App\Result $result): \phpbu\App\Backup\Source\Status
    {
        $status = new App\Backup\Source\Status();
        $status->uncompressedFile('foo');

        return $status;
    }
}
App\Factory::register('source', 'dummy', MySource::class);
```

`phpbu.xml`
```xml
<phpbu  bootstrap="backup/bootstrap.php">
  <backups>
  <backup name="custom">
      <source type="dummy">
        <option name="dummy" value="dummy"/>
      </source>
      <target dirname="backup/dummy" filename="dummy-%Y%m%d-%H%i.txt"/>
    </backup>
  </backup>
</phpbu>
```

Resulting output
```
phpbu 5.0.9 by Sebastian Feldmann and contributors.

Time: 120 ms, Memory: 4.00MB

OK (1 backup, 0 checks, 0 crypts, 0 syncs, 0 cleanups)
```

**Comment by sebastianfeldmann** [comment]
Can you reproduce this?

What kind of installation are you using?
Composer or PHAR?

Which php version are you using?

Windows Linux MacOS?

**Comment by tonirenner** [comment]
Hi,
i use the PHAR on a linux (ubuntu/centos). I figured out that when i do somthing like this

```xml
<?xml version="1.0" encoding="UTF-8"?>
<phpbu xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
       xsi:noNamespaceSchemaLocation="http://schema.phpbu.de/5.0/phpbu.xsd"
       bootstrap="bootstrap.php">
	<backups>
		<backup name="custom">
			<source type="dummy">
				<option name="dummy" value="dummy"/>
			</source>
			<target dirname="backup/dummy" filename="dummy-%Y%m%d-%H%i.txt"/>
		</backup>
	</backups>
</phpbu>
```
it works.

If i dont specify the bootstrap file (bootstrap="bootstrap.php") like this
```xml
<?xml version="1.0" encoding="UTF-8"?>
<phpbu xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
       xsi:noNamespaceSchemaLocation="http://schema.phpbu.de/5.0/phpbu.xsd">
	<backups>
		<backup name="custom">
			<source type="dummy">
				<option name="dummy" value="dummy"/>
			</source>
			<target dirname="backup/dummy" filename="dummy-%Y%m%d-%H%i.txt"/>
		</backup>
	</backups>
</phpbu>
```
it doesn't work.

But correct me if i'm wrong, but in version 5.0.8 (and before) i dont had to specifiy the bootstrap file in the configuration (https://phpbu.de/manual/current/en/extending-phpbu.html "Example phpbu.xml.dist")?

Thanks
Toni

**Comment by sebastianfeldmann** [comment]
oO I'll check if i crashed the bootstrap option in 5.0.9

**Comment by sebastianfeldmann** [comment]
Indeed I did... fix is coming

**Comment by sebastianfeldmann** [comment]
I've just released version 5.0.10 that fixes the bootstrap cli option issue.

Your hunch was absolutely right by fixing the adapter chicken and egg problem I introduced
another one with the bootstrap handling.

Thank you for reporting and helping to get to the bottom of this
